### PR TITLE
Feat view organizations list and other organization profile

### LINKED ISSUE #52: View Organization List
## Description
As a user,
I need to see the list of all organizations involved in BridgeInTech, 
so that I can go through the list and select which organization I should work with in the mentoring program.,

## Mocks
NA

### Update [Required]
- [ ] must have a list of organizations so that users can view the mentoring program

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 2 mentors

## Estimation
2 - 3 days

### PR DESCRIPTION
**Pull request by mtreacy002**
### Description
Added View Organizations list and Other Organization Profile functionalities

Fixes #52 

### Type of Change:
- Code
- User Interface

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
- Login and go to Organizations page. see list of organizations
Screen Shot 2020-08-18 at 12 07 10 am
- Click on an organization from the list and see that Organization Profile page
Screen Shot 2020-08-18 at 9 46 56 am
**Note**
Notice the different Navbar options when you login as user who's representing organization or not.

As a represenative of an organization, you'll see the Navbar below:
Screen Shot 2020-08-18 at 10 01 46 am

Whereas if you don't represent an organization, you'll see the below Navbar:
Screen Shot 2020-08-18 at 10 01 26 am

### Checklist:
- [ ] My PR follows the style guidelines of this project
- [ ] I have performed a self-review of my own code or materials

**Code/Quality Assurance Only**
- [ ] My changes generate no new warnings
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [ ] New and existing unit tests pass locally with my changes

**Additional Note**
To test this features, you should only use the BIT backend server from branch [PR#112](https://github.com/anitab-org/bridge-in-tech-backend/pull/112) as well as MS backend server for BIT ([ms-backend-server](https://github.com/mtreacy002/mentorship-backend/tree/ms-backend-server))

### CODE REVIEW

**Comment by mtreacy002** [comment]
Update @anitab-org/bridgeintech-maintainers . I've completed the View list of Organizations functionality. Now I'm working on redirecting to Organization Profile page when user click on an organization from this organizations list. @meenakshi-dhanani , I did exactly the same as the one for Members list redirecting to Member Profile page, but for some unknown reason, this Organization Profile redirecting doesn't work. 
At first I thought it was caused by the organization name that is used as part of pathname that might not be url suitable. But even after I use the encodeURI to convert the organization name to URI friendly, the redirecting still doesn't work. Can you please advice how to fix this issue? Thanks

File: src/organizations/Organizations.jsx
Comment (meenakshi-dhanani) [inline src/organizations/Organizations.jsx:73]: this needs to be /organization. that's what is mentioned in the Routes. I tried removing the s, I can see the detail view

Comment (mtreacy002) [reply src/organizations/Organizations.jsx:73]: I see. Thanks for pointing that out, @meenakshi-dhanani . 👍

**Comment by mtreacy002** [comment]
Update @anitab-org/bridgeintech-maintainers . I've fixed the redirecting to View Organization profile from selecting from the list of Organizations (thanks to @meenakshi-dhanani sharp eyes 😁). This PR is now complete. Please give your review. In the meantime I'll continue with the backend Programs functionalities.

**Comment by meenakshi-dhanani** [comment]
Hey @mtreacy002 , I'm confused with one thing in the workflow. How do I create an organization and link it to a member? Because I made a PUT request and there it accepted organization details, what links an organization with a member? I think I'm missing something

**Comment by meenakshi-dhanani** [comment]
Also where is the is_organization_representative bit coming in the auth token?

**Comment by mtreacy002** [comment]
> Also where is the is_organization_representative bit coming in the auth token?

@meenakshi-dhanani . It is used in the Navigation to show MyOrganization item on Navbar for user who Is a representative of an organisation as I showed on the screenshots inside the PR description above. Here's the screenshot of the code

Screen Shot 2020-08-19 at 9 05 00 am

**Comment by mtreacy002** [comment]
> Hey @mtreacy002 , I'm confused with one thing in the workflow. How do I create an organization and link it to a member? Because I made a PUT request and there it accepted organization details, what links an organization with a member? I think I'm missing something

@meenakshi-dhanani , If you have done the review on PR #69 , You'll see that in the description I put the steps to create an organization. Please check it out and let me know if you still confused

**Comment by meenakshi-dhanani** [comment]
One bug here - the context goes outside the box

Screen Shot 2020-08-22 at 7 49 41 PM

**Comment by meenakshi-dhanani** [comment]
Also we need to do something about the naming, profile,portfolio you understand and since you explained to me I understand. It's not very obvious for the user.

Could you call the organization that I'm a representative of - currently the component is being referred to as OrganizationProfile - rename to MyOrganization.
The other organization component currently referred to as OtherOrgProfile can be called as Organization. 

I see some more scope of refactoring, but I think this is the least to make it simpler to understand. 

Refrain from using profile/portfolio it's not very obvious from a user perspective. Maybe refer to organization portfolio as Programs because that's what it is.

**Comment by mtreacy002** [comment]
> Also we need to do something about the naming, profile,portfolio you understand and since you explained to me I understand. It's not very obvious for the user.
> 
> Could you call the organization that I'm a representative of - currently the component is being referred to as OrganizationProfile - rename to MyOrganization.
> The other organization component currently referred to as OtherOrgProfile can be called as Organization.
> 
> I see some more scope of refactoring, but I think this is the least to make it simpler to understand.
> 
> Refrain from using profile/portfolio it's not very obvious from a user perspective. Maybe refer to organization portfolio as Programs because that's what it is.


@meenakshi-dhanani . That is exactly why I prefer to use My My in my original submenu [PR](https://github.com/anitab-org/bridge-in-tech-web/pull/37#discussion_r453304295) and said that using just profile/portfolio will be ambiguous. But it was you who were suggesting against it (using My) and asked to use profile/portfolio instead. 
Screen Shot 2020-08-23 at 12 39 20 am

Can I ask you to be consistent in your review 😁? Especially since we have other things on the plate with only so limited time for the remainder of GSoC, it's not good to go back and forth on the same thing. What do you reckon? Can we keep these naming as they are now and refactor them post GSoC? Or perhaps you can open it for other contributor as a first-time-issue only😉.

**Comment by meenakshi-dhanani** [comment]
Paired with Maya to review the PR. The organizations list and organization detail works! Approving

**Review by meenakshi-dhanani — DISMISSED** [review]

**Comment by meenakshi-dhanani** [comment]
@ramitsawhney27 can you take a look? LGTM!

**Comment by mtreacy002** [comment]
Update @anitab-org/bridgeintech-maintainers . I've just pulled a [fix for Create Organization](https://github.com/anitab-org/bridge-in-tech-web/pull/69#issuecomment-678839526) from PR #69.

**Comment by meenakshi-dhanani** [comment]
Hey I checked this again, it works! I found a bug on the backend though, I'm going to report that as a separate issue, it's not part of this issue. I'm approving this PR.

**Review by meenakshi-dhanani — DISMISSED** [review]

**Comment by mtreacy002** [comment]
@meenakshi-dhanani, can you please merge this PR to develop since @foongminwong has approved it. After this one gets merged, I'll pull it to the next PR (otherwise there'll be conflict again once this one gets merged)

**Comment by meenakshi-dhanani** [comment]
> @meenakshi-dhanani, can you please merge this PR to develop since @foongminwong has approved it. After this one gets merged, I'll pull it to the next PR (otherwise there'll be conflict again once this one gets merged)

@mtreacy002 I have already approved this PR. Waiting for another approval

**Comment by mtreacy002** [comment]
Ou, sorry, I thought @foongminwong had approved it, it must be on another PR then.. 😁

**Comment by mtreacy002** [comment]
@anitab-org/bridgeintech-maintainers , Can one other mentor review and approve this PR, please? We want to be able to demo all the completed features on the final demo. However, the demo is limited to code that already get merged to develop. So, if all these 3 last PRs on Frontend can be merged and ready for the demo, that'll be great. Thanks

**Comment by mtreacy002** [comment]
Update @anitab-org/bridgeintech-maintainers , I've just pulled the current branch on develop to this PR. @meenakshi-dhanani , can you please reprove this PR. Thanks
@foongminwong / @ramitsawhney27 / @isabelcosta , can anyone of you please give your review on this PR? It's been 3 days since @meenakshi-dhanani given her approval and we only have tomorrow left to approve and merge everything to develop branch for the final demo. Your help is greatly appreciated. Thanks

**Review by foongminwong — APPROVED** [review]
The changes made in this PR were tested locally. Following are the results:

1. Code review - Done

2. All possible responses were tested as below:

   * _View Organizations List and other organization profile_  
    _Screenshot/gif/url_:  

      test-bit-web-pr-70-view orgs-list-and-other org-profile


    _Expected Result_:  As a user, I should be able to view organization list and other org profile.
    _Actual Result_: Same as expected.

3. OS Version: Windows 10

**Review by meenakshi-dhanani — APPROVED** [review]

**Pull request merged**
